app/helpers/textual_mixins/textual_devices.rb

```diff
@@ -1,129 +1,130 @@
-require 'textual_mixins/device'
-module TextualMixins::TextualDevices
-  def textual_devices
-    devices = devices_details
-    h = {:label    => _("Devices"),
-         :icon     => "fa fa-hdd-o",
-         :explorer => true,
-         :value    => devices.blank? ? _("None") : devices.length}
-    if devices.present?
-      h[:title] = _("Show VMs devices")
-      h[:link]  = url_for_only_path(:action => 'show', :id => @record, :display => 'devices')
+module TextualMixins
+  module TextualDevices
```

Comment:
Can you not have `module TextualMixins::TextualDevices` with zeitwerk?

Reply:
I think you can.  I'll have to try that next.

Reply:
Ah, so loading works but when I run the tests, [lookups like this](https://github.com/ManageIQ/manageiq-ui-classic/blob/cb7e04bf3d4dbd3933d2174ab75f4e6aefa7ab00/app/helpers/textual_mixins/textual_devices.rb#L26) fail.

This is why I removed the collapsed namespace.  I think this format is better than changing the 4 `Device.new` calls to `TextualMixins::Device.new`.

```
  10) TextualMixins::TextualDevices#devices_details with a hardware
      Failure/Error: Device.new(_("Processors"), description, nil, 'pficon pficon-cpu')

      NameError:
        uninitialized constant TextualMixins::TextualDevices::Device
        Did you mean?  TextualMixins::Device
      # ./app/helpers/textual_mixins/textual_devices.rb:25:in `processor_description'
      # ./app/helpers/textual_mixins/textual_devices.rb:122:in `devices_details'
      # ./spec/helpers/textual_mixins/textual_devices_spec.rb:8:in `block (3 levels) in <top (required)>'
      # ./spec/helpers/textual_mixins/textual_devices_spec.rb:17:in `block (4 levels) in <top (required)>'
      # /Users/joerafaniello/.gem/ruby/3.0.4/gems/webmock-3.18.1/lib/webmock/rspec.rb:37:in `block (2 levels) in <main>'

Finished in 1.04 seconds (files took 7.16 seconds to load)
14 examples, 10 failures

Failed examples:

rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:64 # TextualMixins::TextualDevices#disks_attributes with hdd with size_on_disk and percent provisioned collected (AZURE)
rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:50 # TextualMixins::TextualDevices#disks_attributes with hdd with no size_on_disk collected (AZURE)
rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:49 # TextualMixins::TextualDevices#disks_attributes with hdd with no size_on_disk collected (AZURE)
rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:37 # TextualMixins::TextualDevices#disks_attributes with hdd hardware
rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:97 # TextualMixins::TextualDevices#network_attributes with model type parsed
rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:116 # TextualMixins::TextualDevices#network_attributes with dvswitch and dvportgroup
rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:102 # TextualMixins::TextualDevices#network_attributes without vswitch and portgroup
rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:109 # TextualMixins::TextualDevices#network_attributes with vswitch and portgroup
rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:92 # TextualMixins::TextualDevices#network_attributes with network hardware
rspec ./spec/helpers/textual_mixins/textual_devices_spec.rb:17 # TextualMixins::TextualDevices#devices_details with a hardware
```

Reply:
This is the one thing that I hate in Ruby.  Collapsed and uncollapsed modules should be identical ☹️

Reply:
Agreed. The fact that I forgot why this was needed is quite telling.  It doesn't make sense but ruby treats them differently for constant lookup and that's a shame.